addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/handler/SmappeeApplianceHandler.java

```diff
@@ -0,0 +1,104 @@
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.smappee.internal.SmappeeApplianceEvent;
+import org.openhab.binding.smappee.internal.SmappeeService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SmappeeApplianceHandler} is responsible for handling commands and sets the actual status for an appliance.
+ *
+ * @author Niko Tanghe - Initial contribution
+ */
+public class SmappeeApplianceHandler extends AbstractSmappeeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SmappeeApplianceHandler.class);
+
+    ScheduledFuture<?> scheduledJob;
+
+    private String thingId;
+
+    public SmappeeApplianceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        SmappeeService smappeeService = getSmappeeService();
+        if (smappeeService == null || !smappeeService.isInitialized()) {
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            readAppliance(smappeeService);
+        } else {
+            logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+    }
+
+    public void newState(SmappeeApplianceEvent readings) {
+        if (readings != null) {
+            updateState(CHANNEL_APPLIANCE_POWER, new DecimalType(readings.activePower));
+            updateState(CHANNEL_APPLIANCE_LASTUPDATE, new DateTimeType(readings.getTimestamp()));
+        }
+    }
+
+    @Override
+    public void initialize() {
+        thingId = thing.getConfiguration().get(PARAMETER_APPLIANCE_ID).toString();
+
+        updateStatus(ThingStatus.ONLINE);
+
+        // start automatic refresh
+        startAutomaticRefresh();
+    }
+
+    @Override
+    public void dispose() {
+        // stop automatic refresh
+        if (scheduledJob != null) {
+            scheduledJob.cancel(true);
+        }
+    }
+
+    public void startAutomaticRefresh() {
+        SmappeeService smappeeService = getSmappeeService();
+        Runnable runnable = () -> {
+            if (smappeeService != null && smappeeService.isInitialized()) {
+                readAppliance(smappeeService);
+            }
+        };
+
+        scheduledJob = scheduler.scheduleWithFixedDelay(runnable, 0, smappeeService.getConfig().pollingInterval,
+                TimeUnit.MILLISECONDS);
+    }
+
+    private void readAppliance(SmappeeService smappeeService) {
+        SmappeeApplianceEvent readings = smappeeService.getLatestApplianceReading(thingId);
+        if (readings == null) {
+            logger.warn("failed to read to read power consumption for appliance {}", thingId);

```

Comment:
reduce to debug and rather set the Thing status to OFFLINE_COMMUNICATION-ERROR

same for the other handlers

Reply:
OK